### PR TITLE
Separate forward head

### PR DESCRIPTION
**Pull request by vwxyzjn**
This works!

https://beaker.org/ex/01JAZWC6VXP28S76MZ4EZGGQA6/tasks/01JAZWC6W4FW25TJJE3QHC5KB2/job/01JAZWC70YFATM19KXYCG4J58Y

### CODE REVIEW

File: open_instruct/dpo_tune_cache.py [hunk 1]
```diff
@@ -103,7 +104,7 @@ class FlatArguments:
     config_name: Optional[str] = field(
         default=None, metadata={"help": "Pretrained config name or path if not the same as model_name"}
     )
-    dpo_use_paged_optimizer: bool = field(
+    it: bool = field(
```
Comment (natolambert) [inline open_instruct/dpo_tune_cache.py]: why is this called `it`?

File: open_instruct/dpo_tune_cache.py [hunk 2]
```diff
@@ -794,7 +806,8 @@ def load_model():
         collate_fn=DataCollatorForSeq2SeqDPO(tokenizer=tokenizer, model=model, padding="longest"),
         batch_size=args.per_device_train_batch_size,
     )
-
+    print("=============data loaded")
+    print_gpu_stats(init_gpu_memory)
```
Comment (natolambert) [inline open_instruct/dpo_tune_cache.py]: Same print twice

File: open_instruct/dpo_utils.py [hunk 3]
```diff
@@ -170,6 +170,27 @@ def _get_batch_logps(
         return (per_token_logps * loss_mask).sum(-1)
 
 
+def process_batch(
```
Comment (natolambert) [inline open_instruct/dpo_utils.py:173]: This confuses me a bit, but willing to trust. Not sure where its used.

Comment (natolambert) [reply open_instruct/dpo_utils.py:173]: Ah, it's just like "input_ids_chosen"

**Review by natolambert — APPROVED** [review]
LGTM! Comments on cleaning the GPU utiliztion numbers maybe

File: open_instruct/dpo_tune_cache.py (hunk 1 above)
Comment (vwxyzjn) [reply open_instruct/dpo_tune_cache.py]: Not intended. Removed.

File: open_instruct/dpo_tune_cache.py (hunk 2 above)
Comment (vwxyzjn) [reply open_instruct/dpo_tune_cache.py]: removed

File: open_instruct/dpo_utils.py (hunk 3 above)
Comment (vwxyzjn) [reply open_instruct/dpo_utils.py:173]: Yeah. This is written by claude btw XD

**Comment by vwxyzjn** [comment]
image

Alright it works

**Comment by vwxyzjn** [comment]
Merging as is

**Pull request merged**